### PR TITLE
docs: show sources for referenced files

### PR DESCRIPTION
**Pull request by AriPerkkio**
- Fixes https://github.com/stackblitz/tutorialkit/issues/287

### CODE REVIEW

File: docs/tutorialkit.dev/src/content/docs/reference/react-components.mdx [hunk 1]
```diff
@@ -17,6 +18,8 @@ import sourceExampleSimpleEditor from '@components/react-examples/ExampleSimpleE
 import sourceExampleFileTree from '@components/react-examples/ExampleFileTree?raw';
 import sourceExampleCodeMirrorEditor from '@components/react-examples/ExampleCodeMirrorEditor?raw';
 import sourceExampleTerminal from '@components/react-examples/ExampleTerminal?raw';
+import sourceUseTheme from "@components/react-examples/hooks/useTheme?raw";
```
Comment (Nemikolh) [inline docs/tutorialkit.dev/src/content/docs/reference/react-components.mdx:21]: I would use a different version from that file. That file is very specific to our docs.

Here's what we could show for `hooks/useTheme.ts`:

```ts
export function useTheme() {
  return 'light';
}
```

Comment (AriPerkkio) [reply docs/tutorialkit.dev/src/content/docs/reference/react-components.mdx:21]: We could also remove the hook usage completely and just use static `"light"` string as `theme` in the source code. What do you think?

Comment (Nemikolh) [reply docs/tutorialkit.dev/src/content/docs/reference/react-components.mdx:21]: Hmm but then the docs wouldn't look as good when using viewing them in dark theme.

What drawback do you see in showing a different version for `useTheme`? I don't expect that hook to be changed very often.

File: docs/tutorialkit.dev/src/content/docs/reference/react-components.mdx [hunk 2]
```diff
@@ -74,6 +77,17 @@ See a full example on [StackBlitz][stackblitz-example].
 
 ## Components
 
+These examples have references to following files:
```
Comment (Nemikolh) [inline docs/tutorialkit.dev/src/content/docs/reference/react-components.mdx:80]:
```suggestion
These examples have references to the following files:
```

File: docs/tutorialkit.dev/src/content/docs/reference/react-components.mdx (hunk 1 above)
Comment (Nemikolh) [reply docs/tutorialkit.dev/src/content/docs/reference/react-components.mdx:21]: Ah no worries! :raised_hands:

**Review by Nemikolh — APPROVED** [review]
Nice! :star2:

**Pull request merged**